code/src/components/ProductItem.js

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import { useDispatch } from 'react-redux'
+import { useSelector } from 'react-redux'
+import { products } from '../reducers/products'
+
+import './styles/Listed.css'
+
+export const ProductItem = () => {
+  const dispatch = useDispatch()
+  const productItems = useSelector((store) => store.products.item)
+  console.log('ProductItem:', productItems)
+  console.log('Items:', productItems.product)
+
+  // Restarts 
+  const handleClick = () => {
+    return (
+      dispatch(products.actions.restart())
+    )
+  }
+
+  // Catches error
+  if (productItems.status === 0) {
+    return (
+    <div>
+      <button className="restartButton" onClick={handleClick}>
+        Scan a new product
+      </button>
+      <h1>Product not found</h1>
+    </div>
+      )
+  } else {
+
+  //If product found show info
+  return (
+    <section className="productInfo">
+      <button className="restartButton" onClick={handleClick}>
+        Scan a new product
+      </button>
+      {productItems.product && (<div className="productInfoContainer"><h1>Brand:</h1> <h2>{productItems.product.brands}</h2></div>)}
+      {productItems.product && (<div className="productInfoContainer"><h1>Product:</h1> <h2>{productItems.product.product_name}</h2></div>)}
+      {productItems.product && (<div className="productInfoContainer"><h1>Allergens:</h1><h2>
+        {productItems.product.allergens_hierarchy === [] ? productItems.product.allergens_hierarchy.map((allergen) => 
```

Comment:
I find this line a bit confusing. Reading `productItems.product.allergens_hierarchy === []` looks like you're checking if it's an empty array, and then if it is, then `.map` on it, which feels weird if the array is empty. I guess this is really handling a `null` value which comes if there's no `allergens_hierarchy`? There's a method which you can use to check if something is an array - `Array.isArray(productItems.product.allergens_hierarchy)` which could make the intent a little clearer here.

I also feel ternaries in JSX are simpler to read when on multiple lines. So you could do something like this:

```jsx
{Array.isArray(productItems.product.allergens_hierarchy)
  ? productItems.product.allergens_hierarchy.map((allergen) => (
       allergen.replace(/\w+:/, ' ').replace('-', ' ')
     ))
  : 'No allergens found'}
```

This is very much down to personal preference, though. These sort of issues are the wild west of react land 🤠